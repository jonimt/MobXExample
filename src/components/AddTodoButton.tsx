import React, {FC} from 'react';
import {StyleSheet, ViewStyle, Pressable, Text} from 'react-native';

interface IAddTodoButton {
  onPress: () => void;
}

export const AddTodoButton: FC<IAddTodoButton> = ({onPress}) => {
  return (
    <Pressable testID={'AddTodoButton'} style={styles.button} onPress={onPress}>
      <Text>Add Todo</Text>
    </Pressable>
  );
};

const styles = StyleSheet.create({
  button: {
    backgroundColor: 'pink',
    height: 50,
    borderRadius: 25,
    marginTop: 8,
    marginHorizontal: 15,
    justifyContent: 'center',
    alignItems: 'center',
  } as ViewStyle,
});
