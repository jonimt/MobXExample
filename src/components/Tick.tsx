import React, {FC} from 'react';
import {View, StyleSheet, ViewStyle} from 'react-native';

interface ITick {
  completed: boolean;
}

export const Tick: FC<ITick> = ({completed}) => {
  const color = completed ? 'green' : 'red';

  return (
    <View
      testID={'TodoListItemTick'}
      style={[styles.tick, {backgroundColor: color}]}
    />
  );
};

const styles = StyleSheet.create({
  tick: {
    width: 20,
    height: 20,
    borderRadius: 8,
  } as ViewStyle,
});
