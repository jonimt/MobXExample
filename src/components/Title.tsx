import React, {FC} from 'react';
import {StyleSheet, Text, TextStyle} from 'react-native';

interface ITitle {
  title: string;
}

export const Title: FC<ITitle> = ({title}) => {
  return <Text style={styles.title}>{title}</Text>;
};

const styles = StyleSheet.create({
  title: {
    flex: 1,
    textAlign: 'center',
    fontFamily: 'HelveticaNeue-Medium',
    fontSize: 20,
  } as TextStyle,
});
