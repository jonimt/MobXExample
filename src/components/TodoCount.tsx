import React, {FC} from 'react';
import {StyleSheet, Text, TextStyle} from 'react-native';

interface ITodoCount {
  completed: number;
  total: number;
}

export const TodoCount: FC<ITodoCount> = ({completed, total}) => {
  return (
    <Text style={styles.todoCount}>
      {completed}/{total}
    </Text>
  );
};

const styles = StyleSheet.create({
  todoCount: {
    position: 'absolute',
    right: 15,
    fontFamily: 'HelveticaNeue',
    fontSize: 18,
  } as TextStyle,
});
