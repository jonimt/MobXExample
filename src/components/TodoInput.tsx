import React, {FC} from 'react';
import {StyleSheet, TextInput, TextStyle, View, ViewStyle} from 'react-native';

interface ITodoInput {
  value: string;
  onChangeText: (text: string) => void;
}

export const TodoInput: FC<ITodoInput> = ({value, onChangeText}) => {
  return (
    <View style={styles.padding}>
      <TextInput
        style={styles.textInput}
        onChangeText={onChangeText}
        value={value}
        placeholder={'todo'}
      />
    </View>
  );
};

const styles = StyleSheet.create({
  padding: {
    paddingHorizontal: 15,
  } as ViewStyle,
  textInput: {
    width: '100%',
    height: 50,
    paddingHorizontal: 8,
    textAlign: 'center',
    borderRadius: 25,
    borderWidth: 1,
    borderColor: 'black',
  } as TextStyle,
});
