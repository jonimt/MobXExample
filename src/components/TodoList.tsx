import React, {FC, useCallback} from 'react';
import {StyleSheet, ViewStyle, ScrollView} from 'react-native';
import {observer} from 'mobx-react-lite';
import {Todo} from '../store/TodoStore';
import {TodoListItem} from './TodoListItem';

interface ITodoList {
  todos: Todo[];
}

export const TodoList: FC<ITodoList> = observer(({todos}) => {
  const renderItem = useCallback(
    (todo: Todo) => <TodoListItem key={todo.id} todo={todo} />,
    [],
  );

  return (
    <ScrollView
      style={styles.scrollView}
      contentContainerStyle={styles.scrollViewContent}>
      {todos.map(todo => renderItem(todo))}
    </ScrollView>
  );
});

const styles = StyleSheet.create({
  scrollView: {
    flex: 1,
    width: '100%',
  } as ViewStyle,
  scrollViewContent: {
    justifyContent: 'center',
    alignItems: 'center',
  } as ViewStyle,
});
