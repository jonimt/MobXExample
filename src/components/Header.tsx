import React, {FC} from 'react';
import {StyleSheet, View, ViewStyle} from 'react-native';
import {observer} from 'mobx-react-lite';
import {todoStore} from '../store/TodoStore';
import {TodoCount} from './TodoCount';
import {Title} from './Title';

export const Header: FC = observer(() => {
  return (
    <View style={styles.container}>
      <Title title={'Todos'} />
      <TodoCount
        completed={todoStore.completedTodosCount}
        total={todoStore.totalTodos}
      />
    </View>
  );
});

const styles = StyleSheet.create({
  container: {
    width: '100%',
    height: 50,
    justifyContent: 'center',
    alignItems: 'center',
    flexDirection: 'row',
    borderBottomWidth: 1,
    borderBottomColor: 'black',
  } as ViewStyle,
});
