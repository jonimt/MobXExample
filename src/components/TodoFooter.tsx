import React, {FC, useCallback, useState} from 'react';
import {StyleSheet, KeyboardAvoidingView, ViewStyle} from 'react-native';
import {todoStore} from '../store/TodoStore';
import {AddTodoButton} from './AddTodoButton';
import {TodoInput} from './TodoInput';

export const TodoFooter: FC = () => {
  const [todoTask, setTodoTask] = useState('');

  const onChangeText = useCallback((title: string) => setTodoTask(title), []);

  const addTodo = useCallback(() => {
    setTodoTask('');
    todoStore.addTodo(todoTask);
  }, [todoTask]);

  return (
    <KeyboardAvoidingView behavior={'padding'} style={styles.container}>
      <TodoInput value={todoTask} onChangeText={onChangeText} />
      <AddTodoButton onPress={addTodo} />
    </KeyboardAvoidingView>
  );
};

const styles = StyleSheet.create({
  container: {
    width: '100%',
  } as ViewStyle,
});
