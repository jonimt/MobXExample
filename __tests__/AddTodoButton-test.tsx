import React from 'react';
import 'react-native';
import {fireEvent, render} from '@testing-library/react-native';
import renderer from 'react-test-renderer';
import {AddTodoButton} from '../src/components/AddTodoButton';

describe('AddTodoButton testing', () => {
  // Snapshot test
  it('renders correctly', () => {
    const tree = renderer.create(<AddTodoButton onPress={() => {}} />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  // Testing that AddTodoButton components onPress is working
  // and calling function passed as a props
  it('onPress working', () => {
    const mockOnPress = jest.fn();
    const {getByTestId} = render(<AddTodoButton onPress={mockOnPress} />);

    // To make sure we target correct component, we use testID
    // check AddTodoButton Pressable to see we have give it id of AddTodoButton
    const button = getByTestId('AddTodoButton');

    fireEvent(button, 'press');
    expect(mockOnPress).toHaveBeenCalledTimes(1);
  });
});
