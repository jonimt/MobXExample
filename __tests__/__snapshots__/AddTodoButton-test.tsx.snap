// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AddTodoButton testing renders correctly 1`] = `
<View
  accessible={true}
  collapsable={false}
  focusable={true}
  onBlur={[Function]}
  onClick={[Function]}
  onFocus={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "pink",
      "borderRadius": 25,
      "height": 50,
      "justifyContent": "center",
      "marginHorizontal": 15,
      "marginTop": 8,
    }
  }
  testID="AddTodoButton"
>
  <Text>
    Add Todo
  </Text>
</View>
`;
